### ISSUE TITLE
Functions with multiple varargs compete over the parameters

### ISSUE THREAD

**Issue by SirOlaf** [nim-lang/Nim#19332]
When declaring a function that takes multiple vararg parameters, preferably of different types, they will share one address. This means the same vararg array is interpreted with multiple different types, leading to weird coercion.

Example:
```nim
proc thing(a: varargs[string], b: varargs[int]) = 
  echo a
  echo b

thing("a", 0)
```
Output:
```
["a", ""]
[94561930070512, 0]
```

This also means it will deref an invalid location when the int parameter is non-zero

**Comment by xigoi** [comment]
Interestingly, when I run it on my phone, it procuces output like this:

```nim
["a", ""]
[1, 405290270992]
```

Looks like the `1` is the length of the string. Calling `thing("ábçδ", 0)` produces:

```
["ábçδ", ""]
[7, 434698027280]
```

And in Javacript, it seems to produce UTF-8 code points separated by commas. Output of `thing("ábçδ", 0)`:

```
["ábçδ", ""]
[195,161,98,195,167,206,180, 0]
```

**Comment by metagn** [comment]
Asking because this is marked Showstopper: Are multiple varargs really meant to be supported? Can't we disallow them?

**Comment by Araq** [comment]
Yes we can. It's a showstopper because the compiler miscompiles it.

**Comment by Araq** [comment]
Shouldn't block the upcoming releases as it's a very old bug.

**Comment by SirOlaf** [comment]
Looking into this issue, will open a PR soon
